### ISSUE TITLE
Add support for retrieving all pacts for a given provider

### ISSUE THREAD

**Issue by bethesque** [pact-foundation/pact-net#223]
Hi Neil! Rather than specifying one or more pact URLs, the provider name, and a list of consumer version tags can be specified, and the pact-provider-verifier will determine the right pacts to verify. This feature will be required for an upcoming feature release which will allow "pending pacts" to be verified without breaking the build. The `--provider-version-tag` will also be required for that feature. Here are the relevant bits of the CLI documentation.

Give me a yell if you need some clarification.

```
Usage:
  pact-provider-verifier PACT_URL ... -h, --provider-base-url=PROVIDER_BASE_URL

Options:
      [--provider=PROVIDER]
      [--consumer-version-tag=TAG]                                           # Retrieve the latest pacts with   this consumer version tag. Used in conjunction with --provider. May be specified multiple times.
      [--provider-version-tag=TAG]                                           # Tag to apply to the provider   application version. May be specified multiple times.
```

See:

https://github.com/pact-foundation/pact-provider-verifier#usage

**Comment by neilcampbell** [comment]
@bethesque Will try get this in the next couple of days

**Comment by neilcampbell** [comment]
@bethesque So what should be supplied as the `PACT_URL` in this case? Just a base URL to the broker?

**Comment by bethesque** [comment]
Nothing - it's optional. Either you specify a PACT_URL, or you specify the broker URL and the provider name. Though, if you want, you can specify all of them, and it will fetch the pact by URL and dynamically fetch the pacts by name from the broker as well. If you really wanted.

**Comment by neilcampbell** [comment]
We don't actually use the PACT_BROKER_BASE_URL at all. All broker stuff is currently just setup with PACT_URL (which is currently mandatory in the DSL), so I'm going to need to change things a bit by the sounds of it.

**Comment by bethesque** [comment]
Time to make an omelette!

**Comment by bethesque** [comment]
Sorry to hassle you on this Neil, but we've got the new ["pacts for verification" API](https://github.com/pact-foundation/pact_broker/issues/307) and ["pending pacts"](https://github.com/pact-foundation/pact_broker/issues/320) stuff coming out soon that's going to rely on these fields. If you can find the time, I'd really appreciate it.

**Comment by neilcampbell** [comment]
@bethesque Sorry, completely forgot about this one! Will see what I can do.

**Comment by neilcampbell** [comment]
Released in `2.5.0`

**Comment by bethesque** [comment]
Thanks Neil 🙏
